site/content/blog/2020-02-10-maintaining-consistency-in-codebases-with-go-vet.md

```diff
@@ -0,0 +1,181 @@
+---
+title: Maintaining consistency in codebases with Go Vet
+slug: maintaining-consistency-in-codebases-with-go-vet
+date: 2020-02-10
+categories:
+    - "go"
+author: Jesús Espino
+github: jespino
+community: jesus.espino
+---
+
+At Mattermost we are a bit obsessed with the quality of our code, and one of
+the things that we love is consistent and idiomatic Go. In that sense, go vet
+is a great tool, because it checks a lot of things for us (useless assignments,
+wrong number of Printf parameters, correct format in struct tags, etcetera) as
+part of our CI pipeline, helping us and our contributors to write better code.
+
+The main restriction of go vet is that there are a limited number of absolute
+truths about what is right and what is wrong and so the go vet checks are very
+general.
+
+In Mattermost we have a way of doing certain things like logging, test
+assertions or license headers. We love to keep code consistent so we work hard
+to avoid reintroducing old patterns in the code.
```

Comment:
Although `go vet` has great use in common checks, having domain-specific or even company-specific checks is inevitable to maintain a project at our scale. In Mattermost we have a way of doing certain things like logging, test assertions and adding license headers to the source files. We try our best to keep code consistent therefore we work hard to avoid reintroducing old patterns in the code.